src/connections/allContacts.js

```diff
@@ -11,15 +55,29 @@ export const withContacts = connect(
 )
 
 export const withContactsMutations = withMutations(client => ({
-  createContact: attributes =>
-    client.create('io.cozy.contacts', attributes, null, {
+  createContact: async attributes => {
+    const contacts = await findContactsWithSamePhoneOrEmail(attributes)(client)
+    if (contacts.length === 1) {
+      // TODO: create a more complex updated contact if we need to merge deep properties together.
+      const updatedContact = {
+        ...contacts[0],
+        ...attributes
+      }
+      return client.save(updatedContact)
+    } else if (contacts.length > 1) {
+      throw new Error(
+        'Too many contacts found with same email or phone number.'
+      )
+    }
```

Comment:
I don't know if this code should be in a `createOrUpdateContact` or if the `createContact` must always check if a contact with some key already exists (and update it if necessary).